arango/collection.py

```diff
@@ -2006,21 +2011,25 @@ def import_bulk(
         if sync is not None:
             params["waitForSync"] = sync
 
-        request = Request(
-            method="post",
-            endpoint="/_api/import",
-            data=documents,
-            params=params,
-            write=self.name,
-        )
-
         def response_handler(resp: Response) -> Json:
             if resp.is_success:
                 result: Json = resp.body
                 return result
             raise DocumentInsertError(resp, request)
 
-        return self._execute(request, response_handler)
+        results = []
+        for batch in get_batches(documents, batch_size or len(documents)):
+            request = Request(
+                method="post",
+                endpoint="/_api/import",
+                data=batch,
+                params=params,
+                write=self.name,
+            )
+
+            results.append(self._execute(request, response_handler))
+
+        return results[0] if len(results) == 1 else results
```

Comment:
There is a "bug" here. See my comment below in the unit tests.

Let's just keep it super simple and dumb like this:

```
if batch_size is None:
    # original logic
    request = Request(...)
    return self._execute(request, response_handler) 
else:
    # new logic
    results = []
    ....
    return results
```

Reply:
We _could_ do it like that, but [f722eff](https://github.com/ArangoDB-Community/python-arango/pull/207/commits/f722effe4fa8136f49c72c47ab6e328d93cc82d0) accomplishes the same goal without duplicating any `Request` & `self._execute` references.

What do you think? Happy to switch to what was recommended if preferred

Reply:
I don't want to duplicate either, but I think this is one of those cases where trying to save lines of code results in more bug-prone and inefficient outcome. For example we are initializing the `results` list unnecessarily when `batch_size` is None, and it's harder for new devs to understand what's really happening in line 2032. Also I suspect line 2032 could throw an unhandled KeyError when user passes in an empty documents list.

Reply:
I don't want to duplicate either, but I think this is one of those cases where trying to save lines of code results in more bug-prone and inefficient outcome. For example we are initializing the `results` list unnecessarily when `batch_size` is None, and it's harder for new devs to understand what's really happening in line 2032. Also I suspect line 2032 could throw an unhandled KeyError when user passes in an empty documents list.

Reply:
Makes sense, changed as of [c65af2a](https://github.com/ArangoDB-Community/python-arango/pull/207/commits/c65af2add238347c12d990a70bb7fb733a0a0ed7)